src/main/combatLogParser.ts

```diff
@@ -0,0 +1,434 @@
+        console.log(`[CombatLogParser] Stop watching '${pathSpec}'`);
+    }
+
+    /**
+     * Unwatch all paths.
+     */
+    unwatch(): void {
+        Object.keys(this._handlers).forEach(this.unwatchPath.bind(this));
+    }
+
+    /**
+     * Handle a line from the WoW log.
+     */
+     handleLogLine(flavour: string, line: string) {
+        const logLine = new LogLine(line)
+        const logEventType = logLine.type();
+
+        this.emit(logEventType, logLine, flavour);
+    }
+
+    /**
+     * Find and return the flavour of WoW that the log directory
+     * belongs to by means of the '.flavor.info' file.
+     */
+     private getWowFlavour(pathSpec: string): string {
+        const flavourInfoFile = path.normalize(
+            path.join(pathSpec, '../.flavor.info')
+        );
+
+        // If this file doesn't exist, it's not a WoW combat log directory
+        if (!fs.existsSync(flavourInfoFile)) {
+            return 'unknown';
+        }
+
+        const content = fs.readFileSync(flavourInfoFile)
+            .toString()
+            .split("\n");
+
+        return content.length > 1 ? content[1] : 'unknown';
+    }
+
+    /**
+     * Ensure only a single logfile is being watched once one of them starts
+     * receiving data.
+     *
+     * The lock will timeout after 30 seconds if no data has been received.
+     */
+    private lockHandler(handler: CombatLogMonitorHandlerType): boolean {
+        // If it's locked, and not by 'handler', get out.
+        if (this._handlerLock && this._handlerLock !== handler) {
+            return false;
+        }
+
+        // Reset timeout and return
+        if (this._handlerLock === handler) {
+            this.resetLockTimeout();
+            return true;
+        }
+
+        console.log(`[CombatLogParser] Locking path '${handler.path}' for exclusive event processing.`)
+
+        this.emit('DataFirstEvent', handler.wowFlavour);
+
+        this.resetLockTimeout();
+
+        this._handlerLock = handler;
+        return true;
+    }
+
+    /**
+     * Reset the lock/data timeout timer such that it doesn't fire when it
+     * isn't supposed to.
+     */
+    private resetLockTimeout(): void {
+        if (this._handlerLockTimeout) {
+            clearTimeout(this._handlerLockTimeout);
+        }
+
+        this._handlerLockTimeout = setTimeout(() => {
+            console.log(`[CombatLogParser] Unlocking path '${this._handlerLock?.path}' for exclusive event processing since no data received in ${this._options.dataTimeout / 1000} seconds.`)
+            this._handlerLock = undefined;
+            this.emit('DataTimeout', this._options.dataTimeout);
+        }, this._options.dataTimeout);
+    }
+
+    /**
+     * Find and return the most recent file that matches the combat log filename
+     * pattern.
+     */
+    private getLatestLog (pathSpec: string): string | undefined {
+        const globPath = path.join(pathSpec, 'WoWCombatLog*.txt');
+        const logs = glob.sync(globPath)
```

Comment:
Did we not have a "get most recent file" function somewhere else now? Can we share this? 

Ah it's specific for mp4s, but perhaps we can adapt it to take an extension input? See util.ts `getSortedVideos`

If that's not trivial, then don't worry. 

I'd also love to make this async, see later comment and #155

Reply:
Regarding not being async, this actually gets a lot worse with this PR as we open the possibility of the program running 3 "list files" operation per second instead of 1, so the "mildly annoying" jumping we see now could easily become "moderately annoying"

Reply:
We could definitely merge those two. `getSortedVideos(dir: string)` could become `getSortedFiles(dir: string, pattern: string)`. Good idea!

Re: async, yes - as much as possible things should be async. Definitely agree.

Reply:
The only reason to keep it in CombatLogParser, seperate, is that it's a self-contained unit.

I _could_ make an option in the ctor that would take `globFn` or something to give it a method for finding files. 🤔 Will do!

Reply:
This has been fixed!